main.py

```diff
@@ -69,23 +86,91 @@ def draw() -> Shape:
     return target_shape
 
 
+def draw_second_stage() -> Shape:
```

Comment:
This looks like a carbon copy of `draw()`, can you briefly explain why we needed a second method? If we do need a second method, we can be a little more DRY with he method contents so we're not repeating ourselves so much.

Reply:
The second method only generates one image at a time, so we are increasing the attention the subject has to pay, since they now have to look for where the one image is being generated

Reply:
@pedrorribeiro Sweet, I will look at introducing some stage classes where we can easily define the method of generation rather than duplicating code. 👍